### ISSUE TITLE
jsTree delete the created node by calling create_node option when calling data.instance.refresh();

### ISSUE THREAD

**Issue by rahulsharma29** [vakata/jstree#1509]
I am using jsTree to create tree structure of data I am adding certain operation on tree using context_menu option list create, rename, delete. I applied the code for create_node and rename_node.After create_node I call rename_node but if the person cancel the operation I call refresh by calling data.instance.refresh(); to get back the last state but it always deleted the newly created node. Here is the code which I used. please suggest what I am doing wrong```

.

$('#jstree1').jstree({"
                "core" : {"
                "check_callback\" : true,"
                'themes' : {"
                'responsive' : true"
                "}"
                "},"
                "'force_text' : true,"
                "plugins\" : [ \"contextmenu\", \"dnd\",\"wholerow\" ]"
                "}).on('create_node.jstree', function (e, data) {"
                "$.get(\"createNode.action?operation=create_node\", {'baseId' : data.node.id, 'parentId' : data.node.parent, 'categType': data.node.li_attr.catgtype, 'text' : data.node.text, 'nodeSubType':nodeSubTypeVlaue })"
                ".done(function (d) {"
                "data.instance.set_id(data.node, d.id);"
                "data.instance.set_Parameters(data.node, d, data);"
                        "})"
                        ".fail(function () {"
                        "data.instance.refresh();"
                        "});"
                "}).on(\"rename_node.jstree\", function (e, data) {"
                "var previousValue =data.node.text; "
                "$('#contextMenuRename')"
                ".modal({ backdrop: 'static', keyboard: false })"
                ".one('click', '#renameBtn', function (e) {"
                "$.get(\"createNode.action?operation=rename_node\", {'baseId' : data.node.id, 'parentId' : data.node.parent, 'categType': data.node.li_attr.catgtype, 'text' : data.node.text })"
                ".done(function (d) {"
                "$('#contextMenuRename').modal('hide');"
                        "})"
                        ".fail(function () {"
                        "$('#contextMenuRename').modal('hide');"
                        "data.instance.refresh();"
                        "});"
                "}).one('click','#renameCancelBtn,#renmCrossBTN', function(e){"
                "data.instance.refresh();"
                "});"
                "});

```

**Comment by vakata** [comment]
If the node disappears that means that you either are not reading the data directly from the server, or that you failed to persist it. Either way - it is not a problem with jstree but with your data. From what I see you are using HTML as your data, which means calling refresh will revert the tree to what it was when the DOM loaded. Read the docs on hooking up data directly from the server - if that is not what you need - then don't call `.refresh`.

**Comment by rahulsharma29** [comment]
Thanks for the reply. 

Actually I am creating new node. I fallow the flow as I called create_node.jstree for creating node using jstree and then I call createNode.action?operation=create_node\", {'baseId' : data.node.id, 'parentId' : data.node.parent by passing following values to the server to create new node in data base that was done successfully but after receiving the response from the server by ".done(function (d)  I call data.instance.set_id(data.node, d.id);" "data.instance.set_Parameters(data.node, d, data); to set the values received form the server that was also done successfully. but when I called refresh on rename call it clears the created node. what else I can use in place of refresh to get last state of node?
